### ISSUE TITLE
[SMTChecker] ICE because sorts tuple(uint, uint) and struct Item are incompatible

### ISSUE THREAD

**Issue by bshastry** [ethereum/solidity#11065]
One more

```
pragma experimental SMTChecker;
pragma abicoder v1;
struct Item {
uint x;
uint y;
}

contract D {
Item[][][] public items;

function test() public view returns (uint) {
    // The autogenerated getters to not use ABI encoder.
    (uint a, uint b) = this.items(1, 2, 3);
    return a + b;
}
}
// ----
```

throws

```
terminate called after throwing an instance of 'boost::wrapexcept<solidity::smtutil::SMTLogicError>'
  what():  Sorts tuple(uint256,uint256) and struct Item are incompatible
```

**Comment by cameel** [comment]
This happens only when the code contains the `experimental SMTChecker` pragma, right? It's missing in the code snippet.

On 0.8.2 with the pragma I'm getting a similar message but without `terminate`:
```
SMT logic error during analysis:
/solidity/libsmtutil/Z3Interface.cpp(270): Throw in function z3::expr solidity::smtutil::Z3Interface::toZ3Expr(const solidity::smtutil::Expression&)
Dynamic exception type: boost::wrapexcept<solidity::smtutil::SMTLogicError>
std::exception::what: Sorts |tuple(uint256,uint256)| and |struct Item| are incompatible
[solidity::util::tag_comment*] = Sorts |tuple(uint256,uint256)| and |struct Item| are incompatible
```

**Comment by blishko** [comment]
Fixed by #11059.
